src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -25,8 +32,17 @@ internal class FeatureManagementKeyValueAdapter : IKeyValueAdapter
                 throw new FormatException(setting.Key, e);
             }
 
+            foreach (string prefix in _featureFlagPrefixes)
+            {
+                if (featureFlag.Id.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    featureFlag.Id = featureFlag.Id.Substring(prefix.Length);
```

Comment:
If featureflag.Id gets set to empty string, that doesn't break anything does it?

Reply:
Not for feature flags, since we append the `featureflag.Id` to `"FeatureManagement:"` string. 
But this made notice that the same logic for trimming prefix from key-values. If the key is equal to the prefix, users' config will contain a key that is an empty string.

Reply:
Does it cause a problem?

Reply:
No, it silently adds the empty key and its corresponding value to `ConfigurationProvider`. Since the key is actually `string.Empty` and not `null`, we don't get any unhandled exception.

Reply:
Okay, thanks! Should be fine then.